app/services/sign_in/assertion_validator.rb

```diff
@@ -49,6 +53,21 @@ def validate_user_attributes
       end
     end
 
+    def validate_subject
+      if user_identifier.blank?
+        raise Errors::ServiceAccountAssertionAttributesError.new message: 'Assertion subject is not valid'
+      end
+    end
+
+    def validate_timestamps
```

Comment:
can we split this into two functions, `validate_issued_at` (iat), and `validate_expiration`